airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/SchedulerApp.java

```diff
@@ -261,6 +256,15 @@ public static void main(String[] args) throws IOException, InterruptedException
       throw new IllegalStateException("Unable to retrieve Airbyte Version, aborting...");
     }
 
+    TrackingClientSingleton.initialize(
+        configs.getTrackingStrategy(),
+        // todo (cgardens) - we need to do the `#runServer` pattern here that we do in `ServerApp` so that
+        // the deployment mode can be set by the cloud version.
+        new Deployment(DeploymentMode.OSS, jobPersistence.getDeployment().orElseThrow(), configs.getWorkerEnvironment()),
```

Comment:
@jrhizor  i did this heinous thing to get around the merge conflict. Obviously this is not how we want to set deployment mode. do you have thoughts on the right way to inject it? i'm still wrapping my head around the new factory.